src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -38,23 +41,77 @@ import scala.annotation.tailrec
 
 object Homework :
 
+  //solution for booleans
   object `Boolean Operators` :
 
-    def not(b: Boolean): Boolean = ???
+    def not(b: Boolean): Boolean =
+      if (b)
+        false
+      else
+        true
+
+    def and(left: Boolean, right: Boolean): Boolean =

```

Comment:
This solution could be simplified.
Hint: what if `left` is false? Should we check `right`?

Reply:
Since we reproduce "Java" eager operation, then we should prove that if an evaluation of `right` if not necessary then `right` should not be evaluated.
Hint: "call-by-name".